### PR TITLE
fix(provider): return proper repository

### PR DESCRIPTION
**Pull request by dineshsalunke**
check if the entity has tree metadata and return a tree repository if it exists

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?
Currently the providers don't return a TreeRepository for tree entities.

## What is the new behavior?
Now the provider will create and return a tree repository for tree entities

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: lib/typeorm.providers.ts [hunk 1]
```diff
@@ -10,9 +10,13 @@ export function createTypeOrmProviders(
   return (entities || []).map((entity) => ({
     provide: getRepositoryToken(entity, dataSource),
     useFactory: (dataSource: DataSource) => {
-      return dataSource.options.type === 'mongodb'
-        ? dataSource.getMongoRepository(entity)
-        : dataSource.getRepository(entity);
+      const enitityMetadata = dataSource.entityMetadatas.find( meta => meta.target === entity )
```
Comment (micalevisk) [inline lib/typeorm.providers.ts:13]:
```suggestion
      const enitityMetadata = dataSource.entityMetadatas.find((meta) => meta.target === entity)
```

**Comment by Kisama** [comment]
I am waiting this PR is merged

**Comment by micalevisk** [comment]
you can use [`patch-package`](https://www.npmjs.com/package/patch-package) until the merge

I don't have permissions to merge PRs

**Review by micalevisk — APPROVED** [review]

**Comment by kamilmysliwiec** [comment]
LGTM

**Pull request merged**